src/d_main.c

```diff
@@ -860,43 +865,95 @@ static char *GetAutoloadDir(const char *base, const char *iwadname, boolean crea
     return result;
 }
 
-static void PrepareAutoloadPaths (void)
+static void PrepareBase(void)
 {
-    int i;
+    #define BASE_FILE_NAME PROJECT_SHORTNAME".pk3"
+    mz_zip_archive zip_archive = {0};
+
+    char *basefile = M_StringJoin(D_DoomPrefDir(), DIR_SEPARATOR_S,
+                                  BASE_FILE_NAME, NULL);
+
+    boolean result = mz_zip_reader_init_file(&zip_archive, basefile, 0);
+
+    if (!result)
+    {
+        result = mz_zip_reader_init_file(&zip_archive,
+          "../share/" PROJECT_SHORTNAME "/" BASE_FILE_NAME , 0);
+        if (!result)
+        {
+            I_Error("PrepareBasePaths: Failed to open %s", basefile);
```

Comment:
Using `basefile` here isn't quite exact.

Reply:
Thanks for the review. I'm thinking of making a new PR with new functions and a separate PR with graphics conversion/function renaming etc. I also want to work with .zip/.pk3 archives in memory without temporary files.

I assume you are ok with using libsnpg. What do you think about git submodules? I've read some negative comments about them.

Reply:
> I assume you are ok with using libsnpg. 

Yes, I like it and the fact that it plays nicely with miniz.

> What do you think about git submodules? I've read some negative comments about them.

I'm against them. We are mostly importing "static code", and not some vivid external projects like e.g. ZMusic which just happen to be developed in a separate GIT tree. The very meaning of libs like miniz and spng is that they are developed out and do not change frequently anymore.

Reply:
Understood.